themes/ddbasic/template.php

```diff
@@ -953,6 +680,222 @@ function ddbasic_process_page(&$vars) {
  * Adds wrapper classes to the different groups on the ting object.
  */
 function ddbasic_preprocess_ting_object(&$vars) {
+  //
+  // Add tpl suggestions for node view modes.
+  if (isset($vars['elements']['#view_mode'])) {
+    $vars['theme_hook_suggestions'][] = $vars['elements']['#bundle'] . '__view_mode__' . $vars['elements']['#view_mode'];
+  }
+
+  switch ($vars['elements']['#entity_type']) {
+    case 'ting_collection':
+      // Add a reference to the ting_object if it's included in a
+      // ting_collection.
+      foreach ($vars['object']->entities as &$ting_entity) {
+        $ting_entity->in_collection = $vars['object'];
+      }
+      break;
+
+    case 'ting_object':
+
+      $uri_collection = entity_uri('ting_collection', $vars['object']);
+      $vars['ting_object_url_collection'] = url($uri_collection['path']);
+
+      $uri_object = entity_uri('ting_object', $vars['object']);
+      $vars['ting_object_url_object'] = url($uri_object['path']);
+
+      switch ($vars['elements']['#view_mode']) {
+
+        // Teaser.
+        case 'teaser':
+          $vars['content']['group_text']['read_more_button'] = array(
+            array(
+              '#theme' => 'link',
+              '#text' => t('Read more'),
+              '#path' => $uri_object['path'],
+              '#options' => array(
+                'attributes' => array(
+                  'class' => array(
+                    'action-button',
+                    'read-more-button',
+                  ),
+                ),
+                'html' => FALSE,
+              ),
+            ),
+            '#weight' => 9998,
+          );
+
+          if ($vars['object']->is('reservable')) {
+
+            drupal_add_library('system', 'drupal.ajax');
+
+            $vars['content']['group_text']['reserve_button'] = array(
```

Comment:
This is already handled by `ding_reservation_ding_entity_buttons()`. Why do we need to do it again?